### PR TITLE
docs: Simplify menu configuration

### PR DESCRIPTION
**Pull request by davestewart**
This is just a little stitch-in-time PR with some menu config cleanup:

- utility functions replace raw sidebar menu config
- utility functions replace raw head config

This makes it simpler to [grok and update the sidebar menu](https://github.com/davestewart/wxt/blob/feat/docs-cleanup/docs/.vitepress/config.ts#L73) structure:

```ts
sidebar: {
  '/guide/': menuRoot([
    menuGroup('Directory Structure', '/guide/directory-structure/', [
      menuItem('.output/', 'output'),
      menuItem('composables/', 'composables'),
      menuGroup('entrypoints/', '/guide/directory-structure/entrypoints/', [
        menuItem('background', 'background.md'),
        menuItem('bookmarks', 'bookmarks.md'),
        ...
      ]),
      ...
  ]),
}
```

Additionally, some minor style updates:

- main sidebar sections now use correct level 0 (this can be [styled](https://github.com/wxt-dev/wxt/pull/709/files#diff-096f06e4edeeaa293d906d7146e243667e1377d2133e35b9e240525bd79be5b5R88-R94)):<br>CleanShot 2024-06-10 at 13 32 37
- typedoc sidebar config now [shows badges](https://github.com/wxt-dev/wxt/pull/709/files#diff-702f6a4b41fe8e3d3094e575522859877a8cfb040263991f958de4768a6b3f8fR66-R69) for leaf-level collections:<br>CleanShot 2024-06-10 at 13 31 17

I note the original config used an interim `{ items: [ ... ] }` node to restyle the menu, but we can just do that in `theme/custom.css` if you want to tweak it further.

### CODE REVIEW

File: docs/.vitepress/config.ts
Comment (aklinker1) [inline docs/.vitepress/config.ts:39]: Don't really like this third argument here, it's not clear just from looking at it what the third argument is doing. Not a huge problem, since it will probably never be used again.. but I would prefer two functions, `metaProperty` and `metaName` or just have the key be the first argument, like `meta('name', 'twitter:card', 'summary_large_image')`. Or make the third argument an object so the boolean can be labeled: `meta('twitter:card', 'summary_large_image', { useName: true })`

File: docs/.vitepress/utils/menus.ts
Comment (aklinker1) [inline docs/.vitepress/utils/menus.ts:68]: Didn't know this was interpreted as innerHTML, nice!

File: docs/.vitepress/config.ts
Comment (aklinker1) [inline docs/.vitepress/config.ts:122]: Mistake from my previous PR.

```suggestion
          menuItem('utils/', 'utils'),
```

File: docs/.vitepress/config.ts
Comment (aklinker1) [inline docs/.vitepress/config.ts:119]:
```suggestion
          menuGroup('public/', '/guide/directory-structure/public/', [
```

**Review by aklinker1 — CHANGES_REQUESTED** [review]
I like the appearance of the sidebar much more now!

Couple of bugs I noticed:
1. The `guide/directory-structure/public/locales` link is broken. It takes me to `/publicpublic/locales.html` instead of `/guide/directory-structure/public/locales.html`
   Screenshot 2024-06-10 at 10 21 18 AM
2. Can't access `/guide/directory-structure/public/index.html` anymore. the menu item wasn't collapsible before, so you could click on it. Now I can only expand/collapse it's children. I'll let you decide how to add it back. This is likely one of the pages with duplicate content that will be moved, so it's fine if it's a temporary solution.

File: docs/.vitepress/utils/menus.ts
Comment (davestewart) [reply docs/.vitepress/utils/menus.ts:68]: Nor did I, I just poked to see what came out the other side!

File: docs/.vitepress/config.ts
Comment (davestewart) [reply docs/.vitepress/config.ts]: So it seems that as long as the parent config has leading and trailing slashes, that the full path will be properly built.

I don't fully understand how the theme is doing this, but it seems that the following picks up the parent `/guide/directory-structure/`:

https://github.com/wxt-dev/wxt/pull/709/files#diff-3def678deb1b1d5a53948eb8491817dde4dc881e032e785fc61cc93d334eefcdR119-R121

```ts
        menuGroup('Directory Structure', '/guide/directory-structure/', [
          ...
          menuItem('public/', 'public/', [
            menuItem('_locales/', 'public/locales'),
          ]),
```

I actually looked to do something within the helpers, but it seems it's not actually needed.

**Comment by davestewart** [comment]
FWIW I find GitHub's resolve changes UI pretty clunky; I'm still not sure if I've clicked all the right buttons, or if the "changes requested" heading is indicating I still need to work, or you need to do work, or both of us need to do work!

LMK

**Review by aklinker1 — APPROVED** [review]
If it says requested changes, it's because I blocked the merging of the PR until I approve it. Individual suggestions in comments don't effect that status. So just re-requesting my review was the right thing to do!

Everything looks good now, thanks

**Pull request merged**
